src/components/Gallery/Gallery.js

```diff
@@ -1,94 +1,85 @@
-import React from 'react';
+// @ts-check
+import React, { useState, useContext, useMemo } from 'react';
 import PropTypes from 'prop-types';
-import Carousel, { Modal, ModalGateway } from 'react-images';
-import { withTranslationContext } from '../../context';
+import { TranslationContext } from '../../context';
+import ModalWrapper from './ModalWrapper';
 
 /**
- * Gallery - displays up to 6 images in a simple responsive grid with a lightbox to view the images.
+ * Gallery - displays up to 4 images in a simple responsive grid with a lightbox to view the images.
  * @example ../../docs/Gallery.md
- * @extends PureComponent
+ * @typedef {import('types').GalleryProps} Props
+ * @type React.FC<Props>
  */
-class Gallery extends React.PureComponent {
-  static propTypes = {
-    images: PropTypes.arrayOf(
-      PropTypes.shape({
-        /** Url of the image */
-        image_url: PropTypes.string,
-        /** Url of thumbnail of image */
-        thumb_url: PropTypes.string,
-      }),
-    ),
-  };
+const Gallery = ({ images }) => {
+  const [index, setIndex] = useState(0);
+  const [modalOpen, setModalOpen] = useState(false);
+  const { t } = useContext(TranslationContext);
 
-  state = {
-    modalIsOpen: false,
-    currentIndex: 0,
+  /**
+   * @param {number} selectedIndex Index of image clicked
+   */
+  const toggleModal = (selectedIndex) => {
+    if (modalOpen) {
+      setModalOpen(false);
+    } else {
+      setIndex(selectedIndex);
+      setModalOpen(true);
+    }
   };
 
-  toggleModal = (index) => {
-    this.setState((state) => ({
-      modalIsOpen: !state.modalIsOpen,
-      currentIndex: index,
-    }));
-  };
+  const formattedArray = useMemo(
+    () =>
+      images.map((image) => ({
+        src: image.image_url || image.thumb_url,
+      })),
+    [images],
+  );
 
-  render() {
-    const { images, t } = this.props;
-    const formattedArray = images.map((image) => ({
-      src: image.image_url || image.thumb_url,
-    }));
+  const renderImages = images.slice(0, 3).map((image, i) => (
+    <div
+      data-testid="gallery-image"
+      className="str-chat__gallery-image"
+      key={`gallery-image-${i}`}
+      onClick={() => toggleModal(i)}
+    >
+      <img src={image.image_url || image.thumb_url} />
+    </div>
+  ));
 
-    const squareClass = images.length > 3 ? 'str-chat__gallery--square' : '';
+  return (
+    <div
+      className={`str-chat__gallery ${
+        images.length > 3 ? 'str-chat__gallery--square' : ''
+      }`}
+    >
+      {renderImages}
```

Comment:
Does this work or do we need `renderImages()`?

Reply:
returns an array, so `{renderImages}` is fine